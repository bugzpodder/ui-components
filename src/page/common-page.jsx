// @flow
import Button from "@material-ui/core/Button";
import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import React, { type Node, useState } from "react";
import classNames from "classnames";
import styles from "./common-page.module.scss";
import { SideMenu } from "./components/side-menu";
import { SideMenuButton } from "./components/side-menu-button";

type Props = {
  /** Provides the page's Title */
  title?: Node,
  /** Provides for action components to be rendered in the top right corner */
  headerActions?: Array<HeaderAction>,
  /** Provides a subtitle under the `title` */
  subtitle?: Node,
  /**
   * Provides classNames to the page and its sub-components. Options include:
   *
   *  - `root`
   *
   *  - `header` (applied to entire header container which contains headerActions, title, subtitle, and side menu
   *     button)
   *
   *  - `headerActions` (applied to headerActions container)
   *
   *  - `title`
   *
   *  - `subtitle`
   *
   *  - `contentAndMenu` - container around content and side menu.
   *
   *  - `content` - a container inside of contentAndMenu but still around content. This exists to stop the content
   *     from being considered part of a flexbox with the side menu.
   *
   *  - `sideMenu`
   */
  classes?: CommonPageClasses,
  /** Takes a `node` that goes beneath the header. For example, page tabs. */
  subheader?: Node,
  /** Takes a `node` to show on the page */
  children?: Node,
  /** Defines the list of items in the menu. The menu is hidden by default but can be revealed by clicking a hamburger
   *  button to the left of the page. Clicking on each item will scroll to the element with an id matching the key.
   *  If empty, no hamburger icon will appear. */
  menuContents?: Array<MenuItem>,
};

/**
 * `CommonPage` provides a component for a page with a flush card header and an
 * optional side menu. Note: if you are using this component
 * in a new environment, we recommend building a new child component for your
 * environment; take a look at the `LimsPage` component as an example.
 */
export const CommonPage = (props: Props) => {
  const [sideMenuIsExpanded, setSideMenuIsExpanded] = useState(false);

  const toggleSideMenu = () => {
    setSideMenuIsExpanded(sideMenuIsExpanded => !sideMenuIsExpanded);
  };

  const {
    classes = {},
    headerActions = [],
    subtitle = "",
    title = "",
    subheader,
    children,
    menuContents = [],
    ...cardProps
  } = props;
  const mappedHeaderActions = headerActions.map((action, index) => {
    const {
      Component = Button, content = "", id, color = "primary", componentProps, ...otherProps
    } = action;
    return (
      <Component
        key={`header-action-${index}`}
        id={id}
        data-testid={id}
        color={color}
        {...componentProps}
        {...otherProps}
      >
        {content}
      </Component>
    );
  });

  return (
    <div
      className={classNames(classes.root, styles.pageContainer)}
      data-testid="common-page"
    >
      <Card
        data-testid="common-page-header"
        classes={{
          root: classNames(classes.header, styles.stickyHeader, styles.pageHeader),
        }}
        {...cardProps}
      >
        <CardHeader
          title={title}
          titleTypographyProps={{
            variant: "h5",
            "data-testid": "common-page-title",
          }}
          classes={{
            root: styles.cardHeader,
            action: classNames(styles.headerActions, classes.headerActions),
            title: classNames("header-title", classes.title),
            subheader: classes.subtitle,
          }}
          action={mappedHeaderActions}
          subheader={subtitle}
          subheaderTypographyProps={{
            variant: "body2",
            "data-testid": "common-page-subtitle",
          }}
          avatar={
            menuContents.length > 0 && (
            <SideMenuButton
              isExpanded={sideMenuIsExpanded}
              toggleMenu={toggleSideMenu}
            />
            )
          }
        />
        {subheader && subheader}
      </Card>
      <div
        data-testid="common-page-content-and-menu"
        className={classNames(styles.contentAndMenu, classes.contentAndMenu)}
      >
        {menuContents.length > 0 && (
          <SideMenu
            data-testid="common-page-side-menu"
            isExpanded={sideMenuIsExpanded}
            menuContents={menuContents}
            className={classes.sideMenu}
          />
        )}
        <div
          data-testid="common-page-content"
          className={classNames(styles.content, classes.content)}
        >
          {children}
        </div>
      </div>
    </div>
  );
};
