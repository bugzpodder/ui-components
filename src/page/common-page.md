### Examples

```js
const ExampleWrapper = require("../test-utils").ExampleWrapper;
const ReadOnlyTextField = require("@grail/components").ReadOnlyTextField;
const { CommonPage } = require("./index");

const ExampleCommonPage = () => {
  return (
    <CommonPage
      title="Example Common Page"
      subtitle="This is a subtitle"
      headerActions={[
        {
          content: "Button Two",
          color: "primary",
          onClick: () => console.log("Button Two was clicked."),
        },
        {
          content: "Button One",
          color: "primary",
          variant: "contained",
          onClick: () => console.log("Button One was clicked."),
        },
      ]}
    >
      <ReadOnlyTextField>Any content can go inside of CommonPage</ReadOnlyTextField>
    </CommonPage>
  );
};

<ExampleWrapper>
  <ExampleCommonPage />
</ExampleWrapper>;
```

Here is an example with a side menu as well. Note that clicking on the
menu will typically cause the page to scroll to the element whose `id`
matches the `key` given in the object in `menuContents`. However, it
uses CSS anchors and these do not work on styleguidist.

```js
const Alert = require("@grail/components").Alert;
const CommonSwitch = require("@grail/components").CommonSwitch;
const ExampleWrapper = require("../test-utils").ExampleWrapper;
const Fragment = require("react").Fragment;
const { CommonPage } = require("./index");

const ExampleCommonPage = () => {
  return (
    <CommonPage
      title="Example With Menu"
      subtitle="Another subtitle"
      headerActions={[
        {
          Component: CommonSwitch,
          componentProps: {
            label: "CUSTOM ACTION",
            onChange: () => {},
            color: "secondary",
          },
        },
      ]}
      menuContents={[
        {
          label: "Element A",
          key: "elementA",
        },
        {
          label: "Element B",
          key: "elementB",
        },
        {
          label: "Element C",
          key: "elementC",
        },
      ]}
    >
      <div>
        <Alert color="success" message="Element A!" id="elementA" />
        <Alert color="success" message="Element B!" id="elementB" />
        <Alert color="success" message="Element C!" id="elementC" />
      </div>
    </CommonPage>
  );
};

<ExampleWrapper>
  <ExampleCommonPage />
</ExampleWrapper>;
```
