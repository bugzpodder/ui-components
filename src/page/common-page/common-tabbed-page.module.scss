@import "../../styles/breakpoints.scss";

.contentContainer {
  flex-grow: 1;
  overflow-y: auto;
}

.tabs {
  .tab {
    @include media-breakpoint-up(sm) {
      width: 180px;
    }
    @include media-breakpoint-up(md) {
      width: 250px;
    }
  }
}
