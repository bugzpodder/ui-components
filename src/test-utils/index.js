// @flow
export {
  ALPHABET_ANIMALS_DATA,
  ADAMS_DATA,
  COUNTRIES,
  EMU_DATA,
  EXTRA_LARGE_EXAMPLE_TABLE_DATA,
  LARGE_EXAMPLE_TABLE_DATA,
  EXAMPLE_TABLE_DATA,
  GRAIL,
} from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
export { StyleWrapper as ExampleWrapper } from "../style-wrapper";
