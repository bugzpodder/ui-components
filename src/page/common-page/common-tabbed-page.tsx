import React, { ReactNode } from "react";
import classNames from "classnames";
import styles from "./common-tabbed-page.module.scss";
import { CommonPage } from "./common-page";
import {
  CommonTabbedPageClasses,
  HeaderAction,
  PageConfig,
} from "../../types/card";
import { SpinnerOverlay } from "../../spinner-overlay";
import { TabProps } from "@material-ui/core/Tab";
import { TabsComponent } from "./components/tabs-component";

type Props = {
  /**
   * The array of objects used to define the tabs displayed in the header, and their corresponding pages.
   * The `key` property is used to determine which tab is selected.
   */
  pageConfigs: PageConfig[];
  /** Takes the handler used to switch between tabs, based on the tab's value */
  onChangeActiveTab: (x0: string) => any;
  /** The value of the tab header. Used to define which tab is open */
  activeTab: string | false;
  /** Provides the page's Title */
  title?: ReactNode;
  /** Provides for action components to be rendered in the top right corner */
  headerActions?: HeaderAction[];
  /** Provides a subtitle under the `title` */
  subtitle?: ReactNode;
  /**
   * Provides classNames to the page and its sub-components. Options include:
   *
   *  - `root`
   *
   *  - `header` (applied to entire header container which contains headerActions, title, subtitle, and side menu
   *     button)
   *
   *  - `headerActions` (applied to headerActions container)
   *
   *  - `title`
   *
   *  - `subtitle`
   *
   *  - `tabs` (applied to the container of the tab buttons)
   *
   *  - `contentContainer`
   *
   *  - `content`
   *
   *  - `sideMenu`
   */
  classes?: CommonTabbedPageClasses;
  /** Props applied to the `Tabs` container */
  tabProps?: Partial<TabProps>;
  /** Displays an uninteractive loading animation over the page content when true */
  isLoading?: boolean;
};

/**
 * `CommonTabbedPage` provides a component for a page with a flush card header,
 * a tab bar, and an optional side menu. Note: if you are using this component
 * in a new environment, we recommend building a new child component for your
 * environment; take a look at the `TabbedLimsPage` component as an example.
 */
export const CommonTabbedPage: React.FC<Props> = props => {
  const {
    classes = {},
    headerActions = [],
    subtitle = "",
    title = "",
    activeTab,
    onChangeActiveTab,
    pageConfigs,
    tabProps,
    children,
    isLoading = false,
  } = props;
  const { contentContainer, ...commonPageClasses } = classes;
  if (!isLoading && !pageConfigs) {
    throw new Error("pageConfigs is required");
  }
  const selectedTab = pageConfigs.find(tab => tab.key === activeTab);
  if (!isLoading && !selectedTab) {
    console.error("no pageConfig keys match the provided activeTab");
  }
  let PageContent = null;
  let selectedMenuContents = [];
  if (selectedTab && selectedTab.Component) {
    const SelectedComponent = selectedTab.Component;
    selectedMenuContents = selectedTab.menuContents || [];
    PageContent = (
      <div className={classNames(styles.contentContainer, contentContainer)}>
        <SelectedComponent {...(selectedTab.componentProps || {})} />
      </div>
    );
  }
  return (
    <CommonPage
      classes={{ ...commonPageClasses }}
      headerActions={headerActions}
      subtitle={subtitle}
      title={title}
      menuContents={selectedMenuContents}
      subheader={
        <TabsComponent
          activeTab={!isLoading && activeTab}
          onChangeActiveTab={onChangeActiveTab}
          className={classes.tabs}
          pageConfigs={pageConfigs}
          tabProps={tabProps}
        />
      }
    >
      {PageContent}
      {children}
      <SpinnerOverlay isActive={isLoading} />
    </CommonPage>
  );
};
