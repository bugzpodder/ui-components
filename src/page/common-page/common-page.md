### Examples

```js
import { ExampleWrapper } from "../test-utils";
import { ReadOnlyTextField, CommonPage } from "../../";

const ExampleCommonPage = () => {
  return (
    <CommonPage
      title="Example Common Page"
      subtitle="This is a subtitle"
      headerActions={[
        {
          content: "Button Two",
          color: "primary",
          onClick: () => console.debug("Button Two was clicked."),
        },
        {
          content: "Button One",
          color: "primary",
          variant: "contained",
          onClick: () => console.debug("Button One was clicked."),
        },
      ]}
    >
      <ReadOnlyTextField>
        Any content can go inside of CommonPage
      </ReadOnlyTextField>
    </CommonPage>
  );
};

<ExampleWrapper>
  <ExampleCommonPage />
</ExampleWrapper>;
```

Here is an example with a side menu as well. Note that clicking on the
menu will typically cause the page to scroll to the element whose `id`
matches the `key` given in the object in `menuContents`. However, it
uses CSS anchors and these do not work on styleguidist.

```js
import { CommonSwitch, CommonPage } from "../../";
import { ExampleWrapper } from "../test-utils";
import { Alert } from "@material-ui/lab";

const ExampleCommonPage = () => {
  return (
    <CommonPage
      title="Example With Menu"
      subtitle="Another subtitle"
      headerActions={[
        {
          Component: CommonSwitch,
          componentProps: {
            label: "CUSTOM ACTION",
            onChange: () => {},
            color: "secondary",
          },
        },
      ]}
      menuContents={[
        {
          label: "Element A",
          key: "elementA",
        },
        {
          label: "Element B",
          key: "elementB",
        },
        {
          label: "Element C",
          key: "elementC",
        },
      ]}
    >
      <div>
        <Alert serverity="success" id="elementA">
          Element A!
        </Alert>
        <Alert severity="success" id="elementB">
          Element B!
        </Alert>
        <Alert severity="success" id="elementC">
          Element C!
        </Alert>
      </div>
    </CommonPage>
  );
};

<ExampleWrapper>
  <ExampleCommonPage />
</ExampleWrapper>;
```
