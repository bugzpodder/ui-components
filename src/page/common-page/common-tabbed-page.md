### Example

```js
import { Alert } from "@material-ui/lab";
import { CommonTabbedPage } from "./";
import { ExampleBlock, ExampleWrapper } from "../test-utils";
import { useState } from "react";

const ExampleTabbedCard = () => {
  const [activeTab, setActiveTab] = useState("one");
  return (
    <>
      <CommonTabbedPage
        title="Example Tabbed Page"
        subtitle="This is a subtitle"
        activeTab={activeTab}
        onChangeActiveTab={setActiveTab}
        headerActions={[
          {
            content: "Button Two",
            color: "primary",
            onClick: () => console.debug("Button Two was clicked."),
          },
          {
            content: "Button One",
            color: "primary",
            variant: "contained",
            onClick: () => console.debug("Button One was clicked."),
          },
        ]}
        pageConfigs={[
          {
            label: "Tab One",
            key: "one",
            Component: Alert,
            componentProps: {
              color: "success",
              message: "Tab One!",
            },
          },
          {
            label: "Tab Two",
            key: "two",
            Component: Alert,
            componentProps: {
              color: "warning",
              message: "Tab Two!",
            },
          },
        ]}
      />
      <ExampleBlock strongHeader="state " content={activeTab} />
    </>
  );
};

<ExampleWrapper>
  <ExampleTabbedCard />
</ExampleWrapper>;
```

Here is an example with a side menu as well. Note that clicking on the
menu will typically cause the page to scroll to the element whose `id`
matches the `key` given in the object in `menuContents`. However, it
uses CSS anchors and these do not work on styleguidist.

```js
import { useState } from "react";
import { CommonSwitch, CommonTabbedPage } from "../../;
import { ExampleBlock, ExampleWrapper } from "../../test-utils";
import { Alert } from "@material-ui/lab";

const TabOneComponent = () => {
  return (
    <>
      <Alert severity="success" id="elementA">
        Tab One Element A!
      </Alert>
      <Alert severity="success" id="elementB">
        Tab One Element B!
      </Alert>
      <Alert severity="success" id="elementC">
        Tab One Element C!
      </Alert>
    </>
  );
};

const TabTwoComponent = () => {
  return (
    <>
      <Alert severity="warning" id="elementI">
        Tab Two Element I!
      </Alert>
      <Alert severity="warning" id="elementII">
        Tab Two Element II!
      </Alert>
      <Alert severity="warning" id="elementIII">
        Tab Two Element III!
      </Alert>
    </>
  );
};

const ExampleTabbedCard = () => {
  const [activeTab, setActiveTab] = useState("one");
  return (
    <>
      <CommonTabbedPage
        title="Example With Menu"
        subtitle="Another subtitle"
        activeTab={activeTab}
        onChangeActiveTab={setActiveTab}
        headerActions={[
          {
            Component: CommonSwitch,
            componentProps: {
              label: "CUSTOM ACTION",
              onChange: () => {},
              color: "secondary",
            },
          },
        ]}
        pageConfigs={[
          {
            label: "Tab One",
            key: "one",
            Component: TabOneComponent,
            componentProps: {},
            menuContents: [
              {
                label: "Element A",
                key: "elementA",
              },
              {
                label: "Element B",
                key: "elementB",
              },
              {
                label: "Element C",
                key: "elementC",
              },
            ],
          },
          {
            label: "Tab Two",
            key: "two",
            Component: TabTwoComponent,
            componentProps: {},
            menuContents: [
              {
                label: "Element I",
                key: "elementI",
              },
              {
                label: "Element II",
                key: "elementII",
              },
              {
                label: "Element III",
                key: "elementIII",
              },
            ],
          },
        ]}
      />
      <ExampleBlock strongHeader="state " content={activeTab} />
    </>
  );
};

<ExampleWrapper>
  <ExampleTabbedCard />
</ExampleWrapper>;
```
