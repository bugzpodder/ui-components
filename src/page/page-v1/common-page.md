### Examples

```js
import { ExampleWrapper } from "../test-utils";
import { ReadOnlyTextField, CommonPage } from "@grail/components";

const ExampleCommonPage = () => {
  return (
    <CommonPage
      title="Example Common Page"
      subtitle="This is a subtitle"
      headerActions={[
        {
          content: "Button Two",
          color: "primary",
          onClick: () => console.debug("Button Two was clicked."),
        },
        {
          content: "Button One",
          color: "primary",
          variant: "contained",
          onClick: () => console.debug("Button One was clicked."),
        },
      ]}
    >
      <ReadOnlyTextField>Any content can go inside of CommonPage</ReadOnlyTextField>
    </CommonPage>
  );
};

<ExampleWrapper>
  <ExampleCommonPage />
</ExampleWrapper>;
```

Here is an example with a side menu as well. Note that clicking on the
menu will typically cause the page to scroll to the element whose `id`
matches the `key` given in the object in `menuContents`. However, it
uses CSS anchors and these do not work on styleguidist.

```js
import { Alert, CommonSwitch, CommonPage } from "@grail/components";
import { ExampleWrapper } from "../test-utils";

const ExampleCommonPage = () => {
  return (
    <CommonPage
      title="Example With Menu"
      subtitle="Another subtitle"
      headerActions={[
        {
          Component: CommonSwitch,
          componentProps: {
            label: "CUSTOM ACTION",
            onChange: () => {},
            color: "secondary",
          },
        },
      ]}
      menuContents={[
        {
          label: "Element A",
          key: "elementA",
        },
        {
          label: "Element B",
          key: "elementB",
        },
        {
          label: "Element C",
          key: "elementC",
        },
      ]}
    >
      <div>
        <Alert color="success" message="Element A!" id="elementA" />
        <Alert color="success" message="Element B!" id="elementB" />
        <Alert color="success" message="Element C!" id="elementC" />
      </div>
    </CommonPage>
  );
};

<ExampleWrapper>
  <ExampleCommonPage />
</ExampleWrapper>;
```
