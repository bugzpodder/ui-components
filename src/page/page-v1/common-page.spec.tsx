import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { Alert, CommonSwitch } from "../..";
import { CommonPage } from "./common-page";
import { TestWrapper } from "../../test-utils";
import { cleanup, fireEvent, render } from "@testing-library/react";

afterEach(cleanup);

const TestCommonPage: React.FC<any> = props => {
  return (
    <TestWrapper>
      <CommonPage
        {...props}
        subheader={<Alert message="Some alert message" color="success" />}
        value="one"
      >
        Some content here.
      </CommonPage>
    </TestWrapper>
  );
};

test("renders common page", () => {
  const { rerender, container } = render(<TestCommonPage />);
  rerender(
    <TestCommonPage
      title="Test Card"
      subtitle="This is a subtitle"
      headerActions={[
        {
          content: "Test Action",
        },
        {
          Component: CommonSwitch,
          componentProps: {
            label: "CUSTOM ACTION",
            onChange: () => {},
          },
        },
      ]}
    />,
  );
  expect(container).toMatchSnapshot();
});

test("displays a side menu", () => {
  const menuContents = [
    {
      label: "Menu item one",
      key: "one",
    },
    {
      label: "Menu item two",
      key: "two",
    },
  ];

  const { getByTestId } = render(
    <TestCommonPage menuContents={menuContents} />,
  );
  fireEvent.click(getByTestId("toggle-side-menu"));
  expect(getByTestId("side-menu-item-one")).toBeInTheDocument();
  expect(getByTestId("side-menu-item-one")).toHaveAttribute("href", "#one");
});

test("common page classes", () => {
  const { getByTestId } = render(
    <TestCommonPage
      menuContents={[{ key: "test", label: "test" }]}
      classes={{
        root: "test-root",
        header: "test-header",
        headerActions: "test-actions",
        title: "test-title",
        subtitle: "test-subtitle",
        contentAndMenu: "test-container",
        content: "test-content",
        sideMenu: "test-side-menu",
      }}
    />,
  );
  const classes = {
    "common-page": "test-root",
    "common-page-header": "test-header",
    // test headerActions
    "common-page-title": "test-title",
    "common-page-subtitle": "test-subtitle",
    "common-page-content-and-menu": "test-container",
    "common-page-content": "test-content",
    "common-page-side-menu": "test-side-menu",
  };
  Object.keys(classes).forEach(key => {
    expect(getByTestId(key)).toBeInTheDocument();
    expect(getByTestId(key)).toHaveClass(classes[key]);
  });
});
