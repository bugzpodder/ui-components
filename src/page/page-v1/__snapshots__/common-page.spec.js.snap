// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders common page 1`] = `
<div>
  <div
    class="pageContainer"
    data-testid="common-page"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root stickyHeader pageHeader MuiPaper-rounded"
      data-testid="common-page-header"
      value="one"
    >
      <div
        class="MuiCardHeader-root cardHeader"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title header-title MuiTypography-h5 MuiTypography-displayBlock"
            data-testid="common-page-title"
          >
            Test Card
          </span>
          <span
            class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
            data-testid="common-page-subtitle"
          >
            This is a subtitle
          </span>
        </div>
        <div
          class="MuiCardHeader-action headerActions"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Test Action
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="commonSwitch"
          >
            <label
              class="MuiFormControlLabel-root controlLabel"
            >
              <span
                class="MuiSwitch-root"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-114 MuiSwitch-switchBase MuiSwitch-colorPrimary"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <span
                      class="MuiSwitch-thumb"
                    />
                    <input
                      class="PrivateSwitchBase-input-117 MuiSwitch-input"
                      id=""
                      type="checkbox"
                      value=""
                    />
                  </span>
                </span>
                <span
                  class="MuiSwitch-track"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                CUSTOM ACTION
              </span>
            </label>
            
          </div>
        </div>
      </div>
      <div
        class="alertContainer"
        data-testid="alert"
      >
        <div
          class="MuiTypography-root MuiPaper-root MuiPaper-elevation6 MuiSnackbarContent-root success alert-success MuiTypography-body2"
          data-testid="content"
          role="alertdialog"
        >
          <div
            class="MuiSnackbarContent-message"
          >
            <span
              data-testid="success-alert-message"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root alertIcon"
                data-testid="success-alert-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              Some alert message
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="contentAndMenu"
      data-testid="common-page-content-and-menu"
    >
      <div
        class="content"
        data-testid="common-page-content"
      >
        Some content here.
      </div>
    </div>
  </div>
</div>
`;
